packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -436,9 +442,13 @@ export class BaseSnapExecutor {
 
     const request = async (args: RequestArguments) => {
       assertEthereumOutboundRequest(args);
+      const sanitizedArgs = getSafeJson(args);
       this.notify({ method: 'OutboundRequest' });
       try {
-        return await withTeardown(originalRequest(args), this as any);
+        return await withTeardown(
+          originalRequest(sanitizedArgs as unknown as RequestArguments),
```

Comment:
Ideally `RequestArguments` was a valid JSON type, but this is fine for now.

Reply:
It should be, do we still need this cast?

Reply:
IIRC it uses `unknown`, which is not valid JSON.

Reply:
So, I think the only way is to leave this for now?

Without casting:
```
TS2345: Argument of type 'Json' is not assignable to parameter of type 'RequestArguments'.   Type 'null' is not assignable to type 'RequestArguments'.
```

`as RequestArguments`:
```
TS2352: Conversion of type 'string | number | boolean | Json[] | { [prop: string]: Json; } | null' to type 'RequestArguments' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.   Property 'method' is missing in type '{ [prop: string]: Json; }' but required in type 'RequestArguments'.
```